boring-sys/build.rs

```diff
@@ -176,24 +176,25 @@ fn main() {
     use std::path::{Path, PathBuf};
     use std::process::Command;
 
-    if !Path::new("deps/boringssl/CMakeLists.txt").exists() {
-        println!("cargo:warning=fetching boringssl git submodule");
-        // fetch the boringssl submodule
-        let status = Command::new("git")
-            .args(&[
-                "submodule",
-                "update",
-                "--init",
-                "--recursive",
-                "deps/boringssl",
-            ])
-            .status();
-        if !status.map_or(false, |status| status.success()) {
-            panic!("failed to fetch submodule - consider running `git submodule update --init --recursive deps/boringssl` yourself");
+    println!("cargo:rerun-if-env-changed=BORING_BSSL_PATH");
```

Comment:
Hmm, shouldn't we have this for all the other env variables too? Off the top of my head I see `OPT_LEVEL`, `DEBUG`, `CARGO_CFG_TARGET_ARCH`, `CARGO_CFG_TARGET_OS`, ... https://doc.rust-lang.org/cargo/reference/build-scripts.html#rerun-if-env-changed says that env variables set by cargo are automatically rerun, but I suspect there's still a few that aren't being caught.

Happy to do that in a separate PR to avoid making this one too big.

Reply:
:+1: that would be great, thanks!